src/Storages/StorageMerge.cpp

```diff
@@ -1153,6 +1153,63 @@ std::tuple<bool /* is_regexp */, ASTPtr> StorageMerge::evaluateDatabaseName(cons
     return {false, ast};
 }
 
+bool StorageMerge::supportsTrivialCountOptimization() const
+{
+    bool is_support = true;
+
+    auto database_table_iterators = getDatabaseIterators(getContext());
+    for (auto & iterator : database_table_iterators)
+    {
+        while (iterator->isValid())
+        {
+            const auto & table = iterator->table();
+            is_support = table->supportsTrivialCountOptimization();
```

Comment:
It‘s incorrect, should be 
```suggestion
            is_support &= table->supportsTrivialCountOptimization();
```
or return false directly if we found a table does not support trivial count optimization.